kernel_test.go

```diff
@@ -15,6 +16,249 @@ const (
 	success = "\u2713"
 )
 
+const (
+	connectionKey = "a0436f6c-1916-498b-8eb9-e81ab9368e84"
+	sessionID     = "ba65a05c-106a-4799-9a94-7f5631bbe216"
+	transport     = "tcp"
+	ip            = "127.0.0.1"
+	shellPort     = 57503
+	iopubPort     = 40885
+)
+
```

Comment:
To make this file a little more readable I suggest that we shuffle things around a little:

```
consts and vars

//==============================================================================

runTest TestMain and any init

//==============================================================================

Tests

//==============================================================================

Any other test helps including newTestJupyterClient, sendShellRequest, etc.